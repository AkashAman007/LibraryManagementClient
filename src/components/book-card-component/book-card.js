export const BookCard = (props) => {
  console.log(props);
  return<div/>
}