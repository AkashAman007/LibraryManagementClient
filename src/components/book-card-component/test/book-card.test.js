import React from 'react';
import {render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import  { BookCard }  from "../book-card";

afterEach(cleanup);

it("library book card rendered", () => {
    const book = {
        author: "Author Name",
        imageLink: "imageLink",
        title: "Title of Book",
        availableQty: 10,
        description: "Book Description"
    }
    const tree = render(<BookCard book = {book} onBorrowBook={()=>{}}/>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("user borrowed book rendered", () => {
    const book = {
        author: "Author Name",
        imageLink: "imageLink",
        title: "Title of Book",
        availableQty: 10,
        description: "Book Description"
    }
    const tree = render(<BookCard book = {book} onReturnBook={()=>{}} userListing={true}/>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("library book card shows Available Qty", () => {
    const availableBookQty = 10;
    const book = {
        author: "Author Name",
        imageLink: "imageLink",
        title: "Title of Book",
        availableQty: availableBookQty,
        description: "Book Description"
    }
    const tree = render(<BookCard book = {book} onReturnBook={()=>{}}/>)
    const availableQtyText = tree.queryByText("Available Qty.");
    expect(availableQtyText).toBeTruthy;
    const availableBookQtyElement = tree.queryByText(availableBookQty);
    expect(availableBookQtyElement).toHaveClass("badge badge-info");
});

it("user borrowed book card does not show Available Qty", () => {
    const book = {
        author: "Author Name",
        imageLink: "imageLink",
        title: "Title of Book",
        availableQty: 10,
        description: "Book Description"
    }
    const tree = render(<BookCard book = {book} onReturnBook={()=>{}} userListing={true}/>)
    const availableQtyText = tree.queryByText("Available Qty");
    expect(availableQtyText).toBeNull;
});

it("library book card has borrow button", () => {
    const availableBookQty = 10;
    const book = {
        author: "Author Name",
        imageLink: "imageLink",
        title: "Title of Book",
        availableQty: availableBookQty,
        description: "Book Description"
    }
    const tree = render(<BookCard book = {book} onReturnBook={()=>{}}/>)
    const borrowTextElement = tree.queryByText("Borrow");
    expect(borrowTextElement).toBeTruthy;
    expect(borrowTextElement).toHaveClass('btn');
});

it("user book card has return button", () => {
    const availableBookQty = 10;
    const book = {
        author: "Author Name",
        imageLink: "imageLink",
        title: "Title of Book",
        availableQty: availableBookQty,
        description: "Book Description"
    }
    const tree = render(<BookCard book = {book} onReturnBook={()=>{}} userListing={true}/>)
    const borrowTextElement = tree.queryByText("Return");
    expect(borrowTextElement).toBeTruthy;
    expect(borrowTextElement).toHaveClass('btn');
});