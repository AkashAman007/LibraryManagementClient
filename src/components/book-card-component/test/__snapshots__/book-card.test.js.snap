// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library book card rendered 1`] = `
<DocumentFragment>
  <div
    class="book-card card border-primary"
  >
    <img
      class="card-img-top"
      src="imageLink"
    />
    <div
      class="card-body"
    >
      <div
        class="mb-4 h3"
      >
        Title of Book
      </div>
      <div
        class="mb-4 card-subtitle h6"
      >
        <span
          class="text-muted"
        >
          Author Name
        </span>
        <span
          class="float-right h5"
        >
          Available Qty. 
          <span
            class="badge badge-info"
          >
            10
          </span>
        </span>
      </div>
      <p
        class="mb-4 card-text"
      >
        Book Description
      </p>
    </div>
    <div
      class="card-footer"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        type="button"
      >
        Borrow
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`user borrowed book rendered 1`] = `
<DocumentFragment>
  <div
    class="book-card card border-primary"
  >
    <img
      class="card-img-top"
      src="imageLink"
    />
    <div
      class="card-body"
    >
      <div
        class="mb-4 h3"
      >
        Title of Book
      </div>
      <div
        class="mb-4 card-subtitle h6"
      >
        <span
          class="text-muted"
        >
          Author Name
        </span>
      </div>
      <p
        class="mb-4 card-text"
      >
        Book Description
      </p>
    </div>
    <div
      class="card-footer"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        type="button"
      >
        Return
      </button>
    </div>
  </div>
</DocumentFragment>
`;
